packages/connectors/src/apps/dropbox/actions/upload-file/action.ts

```diff
@@ -0,0 +1,93 @@
+import { Readable } from "node:stream";
+import path from "path";
+import { ReadableStream as WebReadableStream } from "node:stream/web";
+
+import { Dropbox, files } from "dropbox";
+import { fetch as undiciFetch } from "undici";
+
+import { defineAction } from "@rollout/framework";
+
+import { DropboxCredential } from "../../auth";
+import { DROPBOX_ROOT_LEVEL } from "../../constants";
+import { inputParamsSchema } from "./input";
+import { extractFilename } from "../../../../utils/extract-file-name";
+
+const LIMIT_150_MB_IN_BYTES = 157286400;
+
+export const action = defineAction<DropboxCredential>()({
+  name: "Upload file",
+  inputParamsSchema,
+  async execute({ credential, resolvedInputParams }) {
+    const { fileUrl, folder, customFileName, overwrite } = resolvedInputParams;
+
+    const dropbox = new Dropbox({ accessToken: credential.data.accessToken });
+
+    const sharedLinkMetadata = await dropbox.sharingGetSharedLinkMetadata({
+      url: folder,
+    });
+
+    if (sharedLinkMetadata.status !== 200) {
+      throw new Error("Dropbox folder selected no longer exist");
+    }
+
+    const response = await undiciFetch(fileUrl);
+    const { headers } = response;
+
+    if (response.status !== 200) {
+      throw new Error(`Failed to get '${fileUrl}' (${response.status})`);
+    }
+
+    if (response.body == null) {
+      throw new Error("Response has no body");
+    }
+
+    const originalFileName = extractFilename(fileUrl, response);
+    const originalExt = path.extname(originalFileName);
+    const uploadFileName =
+      customFileName != null
+        ? `${customFileName}${
+            customFileName.endsWith(originalExt) ? "" : originalExt
+          }`
+        : originalFileName;
+
+    const contentLength = headers.get("content-length") ?? null;
+
+
+    if (contentLength == null) {
+      throw new Error("Could not determine file size");
+    }
+
+    const byteSize = parseInt(contentLength, 10);
+
+    if (byteSize > LIMIT_150_MB_IN_BYTES) {
+      throw new Error("Upload size exceeded");
+    }
+
+  const readable = Readable.fromWeb(response.body as WebReadableStream);
+
+    const dropboxPath: string =
+      sharedLinkMetadata.result.path_lower === DROPBOX_ROOT_LEVEL
+        ? `/${uploadFileName}`
+        : `${sharedLinkMetadata.result.path_lower}/${uploadFileName}`;
+
+    const uploadArgs: files.UploadArg = {
+      path: dropboxPath,
+      contents: readable,
+    };
+
+    if (overwrite === "true") {
+      uploadArgs.mode = {
+        ".tag": "overwrite",
+      };
+    }
+
+    const { result, status } = await dropbox.filesUpload(uploadArgs);
```

Comment:
would prefer to just put this inline, something like this:
```suggestion
    const { result, status } = await dropbox.filesUpload({
      path: dropboxPath,
      contents: readable,
      mode: overwrite === "true" ? { ".tag": "overwrite" } : undefined
    });
```